docker/run-docker

```diff
@@ -0,0 +1,136 @@
+#
+#   Run this file from the root of the repository to build and run a PWA container
+#
+####################################################################################
+
+ENVFILE=./.env
+HOSTSFILE=/etc/hosts
+HOSTPRESENT=false
+
+main () {
+  # copy environment variables to .env in root for docker-compose to consume for build
+  cp ./docker/.env.docker $ENVFILE 
+  . $ENVFILE
+
+  DOMAIN=$PWA_STUDIO_PUBLIC_PATH
+  HOSTMAP="127.0.0.1 $DOMAIN"
+
+  add_domain_to_hostfile
+
+  generate_certificate
+
+  start_docker
+}
+
+add_domain_to_hostfile () {
+  # handle if a domain was not configured
+  if [ "$HOSTMAP" == "127.0.0.1 " ]; then
+    message "Domain not found.\nPlease provide a custom domain in docker/.env.docker under the key:\nPWA_STUDIO_PUBLIC_PATH"
+    exit 1
+  fi
+  
+  # check to see if the hostfile already contains the hostmap/domain
+  while read -r var; do 
+    if [ "$var" == "$HOSTMAP" ]; then
+      HOSTPRESENT=true
+    fi
+  done < $HOSTSFILE
+
+  # add the domain if it is not present in the hostfile, otherwise don't
+  if [ $HOSTPRESENT == false ]; then
+    message "Please provide your password, if requested, to modify $HOSTSFILE"
+    echo $HOSTMAP | sudo tee -a $HOSTSFILE
+
+    # handle error with adding to the hostfile
+    if [ $? == 0 ]; then
+      message "Added $HOSTMAP configured in $ENVFILE to $HOSTSFILE"
+    else 
+      message "Host was not added to $HOSTSFILE.\nPlease enter your password to add $HOSTMAP to your hostsfile."
+      exit $?
+    fi
+  else 
+    message "$HOSTSFILE already configured with $HOSTMAP"
+  fi
+}
+
+generate_certificate () {
+  # make docker/certs folder if one does not already exist
+  [ -d ./docker/certs ] || mkdir ./docker/certs
+
+  # generate a new self-signed certificate if one does not already exist for $DOMAIN
+  if [ -f ./docker/certs/$DOMAIN.crt ]; then
+    message "SSL certificate for $DOMAIN found."
+    return
+  else
+    message "Generating SSL certificate.\nProvide credentials, if requested, to trust the certificate."
+  fi
+
+  # generate an openssl config for creating a certificate signing request (CSR)
+  OpenSSLConf=./docker/certs/"$DOMAIN"-openssl.cnf
+
+  cat >$OpenSSLConf <<EOL
+[req]
+distinguished_name = req_distinguished_name
+x509_extensions = v3_req
+prompt = no
+[req_distinguished_name]
+CN = $DOMAIN
+[v3_req]
+keyUsage = keyEncipherment, dataEncipherment
+extendedKeyUsage = serverAuth
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = *.$DOMAIN
+DNS.2 = $DOMAIN
+EOL
+
+  # create key (no passphrase), CSR, and SSL certificate
+  openssl req \
+  -new \
+  -newkey rsa:2048 \
+  -sha1 \
+  -days 3650 \
+  -nodes \
+  -x509 \
+  -keyout ./docker/certs/$DOMAIN.key \
+  -out ./docker/certs/$DOMAIN.crt \
+  -config $OpenSSLConf
+
+  # add the certificate to the system keychain and trust the certificate
+  sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ./docker/certs/$DOMAIN.crt
```

Comment:
look into adding this to the login keychain rather than the system keychain

Reply:
looked into adding the cert to the login keychain and while I successfully added it to the login keychain file and it was present in the keychain access gui the cert was not being trusted by the browser. despite a lot of google searching I could not get this to work in the login keychain.